File: repos/aspnet-helpdesk/Store.cs
public static class Store
{
    private static Dictionary<int, Agent> _agents = new();
    private static Dictionary<int, Ticket> _tickets = new();
    private static Dictionary<int, Comment> _comments = new();
    private static int _nextAgentId = 1;
    private static int _nextTicketId = 1;
    private static int _nextCommentId = 1;

    public static Agent CreateAgent(string name, string email)
    {
        var agent = new Agent { Id = _nextAgentId++, Name = name, Email = email, Active = true };
        _agents[agent.Id] = agent;
        return agent;
    }

    public static Agent? GetAgent(int id) => _agents.TryGetValue(id, out var a) ? a : null;

    public static List<Agent> ListAgents() => _agents.Values.OrderBy(a => a.Id).ToList();

    public static Ticket CreateTicket(string subject, string priority, int agentId)
    {
        var ticket = new Ticket { Id = _nextTicketId++, Subject = subject, Status = "open", Priority = priority, AgentId = agentId };
        _tickets[ticket.Id] = ticket;
        return ticket;
    }

    public static Ticket? GetTicket(int id) => _tickets.TryGetValue(id, out var t) ? t : null;

    public static List<Ticket> ListTickets() => _tickets.Values.OrderBy(t => t.Id).ToList();

    public static Comment CreateComment(int ticketId, string author, string body, string createdAt)
    {
        var comment = new Comment { Id = _nextCommentId++, TicketId = ticketId, Author = author, Body = body, CreatedAt = createdAt };
        _comments[comment.Id] = comment;
        return comment;
    }

    public static List<Comment> ListCommentsForTicket(int ticketId) =>
        _comments.Values.Where(c => c.TicketId == ticketId).OrderBy(c => c.Id).ToList();

    /// <summary>Test-only hook: wipes all state so each test starts clean.</summary>
    public static void Reset()
    {
        _agents = new();
        _tickets = new();
        _comments = new();
        _nextAgentId = 1;
        _nextTicketId = 1;
        _nextCommentId = 1;
    }
}

File: repos/gin-helpdesk/internal/store/store.go
// Package store holds the in-memory data for gin-helpdesk: agents, tickets,
// and comments. There is no database - state lives in process memory and
// Reset() is a test-only hook to start each test from a clean slate.
package store

import "sort"

type Agent struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
	Email  string `json:"email"`
	Active bool   `json:"active"`
}

type Ticket struct {
	ID       int    `json:"id"`
	Subject  string `json:"subject"`
	Status   string `json:"status"`
	Priority string `json:"priority"`
	AgentID  int    `json:"agentId"`
}

type Comment struct {
	ID        int    `json:"id"`
	TicketID  int    `json:"ticketId"`
	Author    string `json:"author"`
	Body      string `json:"body"`
	CreatedAt string `json:"createdAt"`
}

var (
	agents      = map[int]*Agent{}
	tickets     = map[int]*Ticket{}
	comments    = map[int]*Comment{}
	nextAgentID = 1
	nextTicket  = 1
	nextComment = 1
)

func CreateAgent(name, email string) *Agent {
	a := &Agent{ID: nextAgentID, Name: name, Email: email, Active: true}
	agents[a.ID] = a
	nextAgentID++
	return a
}

func GetAgent(id int) (*Agent, bool) {
	a, ok := agents[id]
	return a, ok
}

func ListAgents() []*Agent {
	out := make([]*Agent, 0, len(agents))
	for _, a := range agents {
		out = append(out, a)
	}
	sort.Slice(out, func(i, j int) bool { return out[i].ID < out[j].ID })
	return out
}

func CreateTicket(subject, priority string, agentID int) *Ticket {
	t := &Ticket{ID: nextTicket, Subject: subject, Status: "open", Priority: priority, AgentID: agentID}
	tickets[t.ID] = t
	nextTicket++
	return t
}

func GetTicket(id int) (*Ticket, bool) {
	t, ok := tickets[id]
	return t, ok
}

func ListTickets() []*Ticket {
	out := make([]*Ticket, 0, len(tickets))
	for _, t := range tickets {
		out = append(out, t)
	}
	sort.Slice(out, func(i, j int) bool { return out[i].ID < out[j].ID })
	return out
}

func CreateComment(ticketID int, author, body, createdAt string) *Comment {
	c := &Comment{ID: nextComment, TicketID: ticketID, Author: author, Body: body, CreatedAt: createdAt}
	comments[c.ID] = c
	nextComment++
	return c
}

func ListCommentsForTicket(ticketID int) []*Comment {
	out := make([]*Comment, 0)
	for _, c := range comments {
		if c.TicketID == ticketID {
			out = append(out, c)
		}
	}
	sort.Slice(out, func(i, j int) bool { return out[i].ID < out[j].ID })
	return out
}

// Reset wipes all in-memory state. Test-only hook: each test file starts
// from an empty store.
func Reset() {
	agents = map[int]*Agent{}
	tickets = map[int]*Ticket{}
	comments = map[int]*Comment{}
	nextAgentID = 1
	nextTicket = 1
	nextComment = 1
}
